src/ResourceManager/EventHub/Commands.EventHub/Cmdlets/EventHub/SetAzureEventHubRegenerateKeys.cs

```diff
@@ -0,0 +1,70 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Management.EventHub.Models;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.EventHub.Commands.Namespace
+{
+    [Cmdlet(VerbsCommon.Set, EventHubRegenerateKeysVerb), OutputType(typeof(ResourceListKeys))]
+    public class SetAzureEventHubRegenerateKeys : AzureEventHubsCmdletBase
+    {
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "The name of the resource group")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroup { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 1,
+            HelpMessage = "Namespace Name.")]
+        [ValidateNotNullOrEmpty]
+        public string NamespaceName { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 2,
+            HelpMessage = "EventHub Name.")]
+        [ValidateNotNullOrEmpty]
+        public string EventHubName { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 3,
+            HelpMessage = "Authorization Rule Name.")]        
+        [ValidateNotNullOrEmpty]
+        public string AuthorizationRule { get; set; }        
+
+        [Parameter(Mandatory = true,
+            Position = 4,
+            ParameterSetName = RegenerateKeysSetName,
+            HelpMessage = "Regenerate Keys - 'PrimaryKey'/'SecondaryKey'.")]
+        [ValidateSet(RegeneKeys.PrimaryKey,
+            RegeneKeys.SecondaryKey,
+            IgnoreCase = true)]
+        [ValidateNotNullOrEmpty]        
+        public string RegenerateKeys { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            var regenKey = new RegenerateKeysParameters(ParsePolicyKey(RegenerateKeys));
+
+            // Get a EventHub List Keys for the specified AuthorizationRule
+            var keys = Client.SetRegenerateKeys(ResourceGroup, NamespaceName, EventHubName, AuthorizationRule, regenKey);
+            WriteObject(keys);
```

Comment:
You should have presentation types that you use as Output types for your cmdlets. This prevents breaking changes in the underlying management library from being breaking changes in your powershell cmdlets.